### PR TITLE
Added cult major loss if Narsie is destroyed.

### PR DESCRIPTION
**Pull request by Buggy123**
See title. If Narsie somehow dies after being summoned, it will be recognized instead of the ending continuing on anyway. This has happened at least once and the inability to stop/change the ending was noted by admins.

This does not add, remove, modify, or care about _how_ Narsie died, it just handles it if it happens.

:cl: 
add: Centcomm has updated their emergency temporal response protocols, and will now correctly stand down if a major threat is destroyed before countermeasures can be deployed.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
How the hell did crew kill an Eldritch Abomination.

Did they BOH bomb it? BSA it? (As BSA's can now kill singulos, and Nar'Sie is a type of singulo.)

**Comment by TheVekter** [comment]
You can still BoH bomb it, which IMO sucks but whatever, that doesn't seem to be the prevailing opinion.

**Comment by necromanceranne** [comment]
You can BoH bomb Nar'sie to null space her

**Comment by 81Denton** [comment]
Direct BSA hit will do the job

**Comment by MCterra10** [comment]
Now I really want to see this happen ahaha

**Comment by GuyonBroadway** [comment]
A singulo will kill nar'sie and a shuttle landing on anything will obliterate it including singularities, Nar'sie and clowns of all types.

**Comment by TetraK1** [comment]
god i hope building BSA becomes an essential part of combating a blood cult

File: code/modules/antagonists/cult/cult.dm [hunk 1]
```diff
@@ -367,18 +368,24 @@
...
+		return -1 // You failed so hard that even the code went backwards.
 	return summoned || completed
 
 /datum/team/cult/proc/check_cult_victory()
 	for(var/datum/objective/O in objectives)
+		if(O.check_completion() == -1)
+			return -1
 		if(!O.check_completion())
 			return FALSE
 	return TRUE
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/cult.dm]: need to undo the T/Fs here, it's no longer a boo

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
I personally would like to see this framed as a more neutral / minor win on cult.

Yeah Narsie got sent back to her realm or whatever but the cult just kicked everyone's butt.

That's unless you plan on making gameplay after the summon that cult can react to, not sure if that was your intention.

**Comment by PKPenguin321** [comment]
Once narsie is summoned the cult gets an infinite army of goons that can hit you twice and perma-disarm you. If the cult manages to let narsie die, that's a big fail on their part

**Comment by ExcessiveUseOfCobblestone** [comment]
that's if they get to you, having someone pop a BoH right as you summoned is not reactable. I guess you could argue that the cult shouldn't have let someone get to that point, but then the same is true on the crew's end: they shouldn't have let the cult get to a point where they could summon. Narsie was changed so the cult is forced to give crew time to react, if you can't beat them here I'm not really sold on why there needs to be a way to force a cult lose after that if you have a fairly popular albeit "late-game" (depending on what shenanigans has gone on) item.

BSA is whatever, you should address a huge static objective before summoning.

**Comment by PKPenguin321** [comment]
>that's if they get to you, having someone pop a BoH right as you summoned is not reactable.

then dont summon if theres a dude with a BoH right next to you? like, stun him first? that's definitely still on you if you let it happen

File: code/modules/power/singularity/narsie.dm [hunk 2]
```diff
@@ -66,34 +66,68 @@
...
 	SSshuttle.lockdown = TRUE
 	sleep(600)
-	if(resolved == FALSE)
-		resolved = TRUE
+	if(QDELETED(GLOB.cult_narsie)) // tres
+		priority_announce("Normalization detected! Abort the solution package!","Central Command Higher Dimensional Affairs", 'sound/misc/notice1.ogg')
+		GLOB.cult_narsie.resolved = TRUE
+		sleep(20)
+		set_security_level("red")
+		SSshuttle.clearHostileEnvironment(GLOB.cult_narsie)
```
Comment (SpaceManiac) [inline code/modules/power/singularity/narsie.dm]: `QDELETED(GLOB.cult_narsie)` is true in this branch, meaning `GLOB.cult_narsie` is likely to be null

Intentionally letting it not be null is a bad idea, you shouldn't hang on to references to something after its Destroy

File: code/modules/antagonists/cult/cult.dm [hunk 3]
```diff
@@ -367,18 +368,24 @@
...
-			return FALSE
-	return TRUE
+			return 0
+	return 1
 
 /datum/team/cult/roundend_report()
 	var/list/parts = list()
 
-	if(check_cult_victory())
+	if(check_cult_victory() == -1) // Epic failure, you summoned your god and then someone killed it.
```
Comment (SpaceManiac) [inline code/modules/antagonists/cult/cult.dm]: You could get away with calling `check_cult_victory()` only once here rather than twice

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by GuyonBroadway** [comment]
Honestly who gives a shit about "muh greentext" and "OMG U STOLE MUH GREENTEXT."   This makes for much more interesting stories that  cults winning quietly after 20 mins of sneaking.

**Comment by Megarop** [comment]
> god i hope building BSA becomes an essential part of combating a blood cult
the crew can barely manage to build a BSA on 3 hour extended rounds you're tripping if you think anyone's gonna build it in the middle of a cult round or that the round is even going to last long enough to research femto manipulators and get 20 bluespace crystals or whatever you need to build a bsa

**Comment by Buggy123** [comment]
This should be done, by the way; I've addressed the requested changes, it just isn't showing as resolved because of how I did it.

File: code/modules/antagonists/cult/cult.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/antagonists/cult/cult.dm]: not booleans (still)

Comment (Buggy123) [reply code/modules/antagonists/cult/cult.dm]: Odd, thought I had got that. It's resolved now.

File: code/modules/power/singularity/narsie.dm (hunk 2 above)
Comment (Buggy123) [reply code/modules/power/singularity/narsie.dm]: This is fixed now.

File: code/modules/antagonists/cult/cult.dm (hunk 3 above)
Comment (Buggy123) [reply code/modules/antagonists/cult/cult.dm]: Done, it's just not showing up.

File: code/modules/antagonists/cult/cult.dm [hunk 4]
```diff
@@ -367,18 +368,25 @@
 	explanation_text = "Summon Nar'Sie by invoking the rune 'Summon Nar'Sie'. <b>The summoning can only be accomplished in [english_list(summon_spots)] - where the veil is weak enough for the ritual to begin.</b>"
 
 /datum/objective/eldergod/check_completion()
+	if(killed)
+		return -1 // You failed so hard that even the code went backwards.
```
Comment (optimumtact) [inline code/modules/antagonists/cult/cult.dm]: make these defines

**Comment by ExcessiveUseOfCobblestone** [comment]
ping me on discord when this is complete

**Pull request merged**